dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,276 @@
+        );
+    }
+
+    /// <summary>
+    /// Transcribe audio file.
+    /// </summary>
+    /// <param name="content">Stream of the audio file</param>
+    /// <param name="executionSettings">The AI execution settings (optional).</param>
+    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Text content from audio content.</returns>
+    public async Task<TextContent> GetTextContentAsync(
+        AudioStreamContent content,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        string uploadUrl = await this.UploadFileAsync(content.Stream, cancellationToken).ConfigureAwait(false);
+
+        var transcriptId = await this.CreateTranscriptAsync(uploadUrl, executionSettings, cancellationToken)
+            .ConfigureAwait(false);
+        var transcript = await this.WaitForTranscriptToProcessAsync(transcriptId, executionSettings, cancellationToken)
+            .ConfigureAwait(false);
+
+        return new TextContent(
+            text: transcript.RootElement.GetProperty("text").GetString(),
+            modelId: null,
+            // TODO: change to typed object when AAI SDK is shipped
+            innerContent: transcript,
+            encoding: Encoding.UTF8,
+            metadata: null
+        );
+    }
+
+    private async Task<string> UploadFileAsync(Stream audioStream, CancellationToken ct)
+    {
+        var url = this.Url("v2/upload");
+        using var content = new StreamContent(audioStream);
+        content.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
+        using var request = new HttpRequestMessage(HttpMethod.Post, url);
+        this.AddDefaultHeaders(request);
+        request.Content = content;
+        using var response = await this.SendWithSuccessCheckAsync(this._httpClient, request, ct).ConfigureAwait(false);
+        var jsonStream = await response.Content.ReadAsStreamAndTranslateExceptionAsync().ConfigureAwait(false);
+        var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+        return json.RootElement.GetProperty("upload_url").GetString()
+               ?? throw new KernelException("Property 'upload_url' expected but not found.");
+    }
+
+    private async Task<string> CreateTranscriptAsync(
+        string audioUrl,
+        PromptExecutionSettings? executionSettings = null,
+        CancellationToken ct = default
+    )
+    {
+        var url = this.Url("v2/transcript");
+        var jsonRequest = new JsonObject();
+        jsonRequest["audio_url"] = audioUrl;
+        if (executionSettings?.ExtensionData is not null)
```

Comment:
You check only extensions data but if it will be AssemblyAIPromoExecutionSettings you ignore polling property?
Take a look how executions settings are loaded in gemini https://github.com/microsoft/semantic-kernel/blob/642747890517a52f6f81a38d35c1ed68dbcf05c5/dotnet/src/Connectors/Connectors.GoogleVertexAI/Settings/GeminiPromptExecutionSettings.cs#L185

Reply:
The polling property isn't being passed to the API, it is used by `AssemblyAIAudioToTextService.WaitForTranscriptToProcessAsync` where the polling of the transcript resource occurs.
This may be inappropriate use of execution settings. Maybe it should be a property of `AssemblyAIAudioToTextService` itself?

Reply:
User will access your service by abstraction IAudioToTextService so all additional properties should be in PromptExecutionSettings.

Reply:
Got it. I ended up moving it to `AssemblyAIAudioToTextServiceOptions` and removed `AssemblyAIPromptExecutionSettings`.